RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -97,13 +97,42 @@ private void ReplaceAssignedByValParameterReferences(IModuleRewriter rewriter, s
             }
         }
 
+
         private void InsertLocalVariableDeclarationAndAssignment(IModuleRewriter rewriter, string localIdentifier)
         {
-            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine
-                + (_target.AsTypeDeclaration is ClassModuleDeclaration ? Tokens.Set + " " : string.Empty)
-                + localIdentifier + " = " + _target.IdentifierName;
+            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine;
+            if (IsBaseTypeContext(_target))
+            {
+                content = content + localIdentifier + " = " + _target.IdentifierName;
+            }
+            else 
+            {
+                //All we can know is that it is not a Base type.  Let VBA determine
+                //the right way to assign the parameter.  The user can simplify it later.
+                string insertIsObjectCheck =
+@"If(IsObject({1})) Then
```

Comment:
So, the code in ObjectVariableNotSet is definitely helpful for most cases.  The test case that lacks an 'AsType' clause though, is not handled correctly.  And, if I add a test case to the ObjectVariableNotSet inspection tests with the following...
```
Public Sub Foo(FirstArg As Long, ByVal arg1)
    arg1 = Range(""A1: C4"")
End Sub
```
the ObjectVariableNotSet inspection does not detect that arg1's assignment requires a "Set" - unless I add "As Range" to parameter arg1's declaration.  It seems that I might need to find the parameter's assignment in the code block, and determine what type the 'RValue' is.  I have yet to find where that has been done anywhere else (hoping for a reference here...).  Another option might be to look for an assignment of the parameter in the code block and see how it is 'done' and replicate it (although this assumes that the user has it right initially).  Other thoughts?